### PR TITLE
Fix `diagram_generator` for Android

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/103272

When attempting to generate a diagram using `diagram_generator` on Android, the app crashes 

This PR makes fixes so `diagram_generator` can run on Android (similar to https://github.com/flutter/flutter/pull/100140)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

Ideally, we would add a test to ensure that this doesn't break again in the future.

File: packages/diagram_generator/android/app/build.gradle [hunk 1]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
```
Comment (blasten) [inline packages/diagram_generator/android/app/build.gradle:29]:
```suggestion
    compileSdkVersion flutter.compileSdkVersion
```

File: packages/diagram_generator/android/app/build.gradle [hunk 2]
```diff
@@ -36,7 +36,8 @@ android {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.example.diagram_generator"
         minSdkVersion 16
-        targetSdkVersion 30
+        targetSdkVersion 31
```
Comment (blasten) [inline packages/diagram_generator/android/app/build.gradle:39]:
```suggestion
        minSdkVersion flutter.minSdkVersion
        targetSdkVersion flutter.targetSdkVersion
```

**Comment by TahaTesser** [comment]
@goderbauer 
Thanks for the review, I've added a smoke integration test.

Do I add the test to https://github.com/flutter/tests?
Similar to https://github.com/flutter/tests/pull/131/files

**Pull request merged**